plugins/woocommerce/src/Internal/Orders/OrderActionsRestController.php

```diff
@@ -130,20 +524,95 @@ public function send_order_details( WP_REST_Request $request ) {
 		WC()->mailer()->customer_invoice( $order );
 
 		$user_agent = esc_html( $request->get_header( 'User-Agent' ) );
-		$note       = sprintf(
+		$messages[] = sprintf(
 			// translators: %1$s is the customer email, %2$s is the user agent that requested the action.
 			esc_html__( 'Order details sent to %1$s, via %2$s.', 'woocommerce' ),
 			esc_html( $order->get_billing_email() ),
 			$user_agent ? $user_agent : 'REST API'
 		);
-		$order->add_order_note( $note, false, true );
+
+		foreach ( $messages as $message ) {
+			$order->add_order_note( $message, false, true );
+		}
 
 		// phpcs:disable WooCommerce.Commenting.CommentHooks.MissingSinceComment
 		/** This action is documented in includes/admin/meta-boxes/class-wc-meta-box-order-actions.php */
 		do_action( 'woocommerce_after_resend_order_email', $order, 'customer_invoice' );
 
 		return array(
-			'message' => __( 'Order details email sent to customer.', 'woocommerce' ),
+			'message' => implode( ' ', $messages ),
 		);
 	}
+
+	/**
+	 * Update the billing email of an order when certain conditions are met.
+	 *
+	 * If the order does not already have a billing email, it will be updated. If it does have one, but `$force` is set
+	 * to `true`, it will be updated. Otherwise this will return an error. This can also return an error if the given
+	 * email address is not valid.
+	 *
+	 * @param WC_Order $order The order to update.
+	 * @param string   $email The email address to maybe add to the order.
+	 * @param bool     $force Optional. True to update the order even if it already has a billing email. Default false.
+	 *
+	 * @return string|WP_Error A message upon success, otherwise an error.
+	 */
+	private function maybe_update_billing_email( WC_Order $order, string $email, ?bool $force = false ) {
+		$existing_email = $order->get_billing_email( 'edit' );
+
+		if ( $existing_email === $email ) {
+			return '';
+		}
+
+		if ( $existing_email && true !== $force ) {
+			return new WP_Error(
+				'woocommerce_rest_order_billing_email_exists',
+				__( 'Order already has a billing email.', 'woocommerce' ),
+				array( 'status' => 500 )
```

Comment:
Would it make more sense to use a 400 status code here instead of 500, since this is a validation case (existing email) rather than an unexpected server error? 🤔

Reply:
Yeah, that makes more sense, good call. Updated in 17e8d4c2fb